diff -pr source/src/ssl.c source-new/src/ssl.c
*** source/src/ssl.c	1969-12-31 19:00:01.000000000 -0500
--- source-new/src/ssl.c	2023-10-05 19:15:46.369551660 -0400
*************** gnutls_session_t ssl_negotiate(struct se
*** 70,75 ****
--- 70,76 ----
  	gnutls_set_default_priority(ssl_ses);
  	gnutls_credentials_set(ssl_ses, GNUTLS_CRD_CERTIFICATE, ssl_cred);
  	gnutls_transport_set_ptr(ssl_ses, (gnutls_transport_ptr_t) (long int) ses->socket);
+ 	gnutls_server_name_set(ssl_ses, GNUTLS_NAME_DNS, ses->session_host, strlen(ses->session_host));
  
  	do 
  	{
